### PR TITLE
Deploy from circle

### PR DESCRIPTION
**Pull request by slipset**
**Replace this placeholder text with a summary of the changes in your PR.
The more detailed you are, the better.**

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [ ] You've updated the [changelog](../blob/master/changes.md) (if adding/changing user-visible functionality)
- [ ] You've updated the [readme](../blob/master/README.md) (if eg adding a new linter)

Thanks!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -90,15 +90,134 @@ jobs:
...
+            - ~/.m2
+          key: v1-dependencies-{{ checksum "project.clj" }}
+      - run:
+         name: Deploy
+         command: |
+           GPG_TTY=$(tty)
+           export EVAL_IN_LEININGEN=true
+           export GPG_TTY
+           echo $GPG_TTY
+           ./circle-maybe-deploy.bb lein deploy clojars
```
Comment (vemv) [inline .circleci/config.yml:212]:
```suggestion
           ./circle-maybe-deploy.bb lein with-profile -dev,+eastwood-plugin deploy clojars
```

Looks necessary?

File: .circleci/config.yml [hunk 2]
```diff
@@ -111,3 +226,10 @@ workflows:
               executor: [openjdk8, openjdk11]
               # Run just the latest version. This saves resources trusts that the main test suite already exercises clojure compat sufficiently:
               clojure-version: ["1.10.3"]
+
+      - deploy:
+          requires:
+            - test
```
Comment (vemv) [inline .circleci/config.yml:232]:
```suggestion
            - test
            - test_third_party_deps:
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -90,15 +90,130 @@ jobs:
...
           name: Prevent any form of caching, for the next step
           command:  git clean -fdx
-                  
+
       - run:
           name: Run test suite that exercises 3rd-party libs
           command: lein with-profile -user,-dev,+test,+warn-on-reflection,+test-3rd-party-deps,+<< parameters.clojure-version >> do clean, test
 
+
+  build:
```
Comment (vemv) [inline .circleci/config.yml:99]: the `build` job looks unused - let's remove it?

File: .circleci/config.yml [hunk 4]
```diff
@@ -111,3 +226,11 @@ workflows:
...
               # Run just the latest version. This saves resources trusts that the main test suite already exercises clojure compat sufficiently:
               clojure-version: ["1.10.3"]
+
+      - deploy:
+          requires:
+            - test
+            - test_third_party_deps
+          filters:
+            tags:
+              only: /Release-.*/
```
Comment (vemv) [inline .circleci/config.yml:236]: We also need

```
            branches:
              ignore: /.*/
```

Comment (slipset) [reply .circleci/config.yml:236]: I don't believe we do. Since this only runs for tags. This way you _can_ release from a branch, like I'm doing now.

Comment (vemv) [reply .circleci/config.yml:236]: Else this will run in every build, which would fail for forks and potentially publish undesired releases?

Comment (slipset) [reply .circleci/config.yml:236]: No. It will only run on tagged commits.

Comment (vemv) [reply .circleci/config.yml:236]: I tried this branch on my personal CI account and it didn't seem so:

image

the `deploy` job would run no matter what (and fail, given I lack the GPG env var)

Comment (vemv) [reply .circleci/config.yml:236]: ...The specific error being:

```
#!/bin/bash -eo pipefail
GNUPGHOME="$HOME/.gnupg"
export GNUPGHOME
mkdir -p "$GNUPGHOME"
chmod 0700 "$GNUPGHOME"

  echo "$GPG_KEY" \
   | base64 --decode --ignore-garbage \
   | gpg --batch --allow-secret-key-import --import

   gpg --keyid-format LONG --list-secret-keys

gpg: keybox '/home/circleci/.gnupg/pubring.kbx' created
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0

Exited with code exit status 2
CircleCI received exit code 2
```

As you hinted over Slack, perhaps we can simply detect if the remote is called `jonase`.

This is exposed as `CIRCLE_PROJECT_USERNAME`

**Pull request merged**